src/utils/timing.rs

```diff
@@ -0,0 +1,118 @@
+use std::time::Instant;
+use tracing::trace;
+
+// Measures the duration of an async operation and logs it using tracing.
+pub async fn measure_operation<T, F, Fut>(operation_name: &str, f: F) -> T
+where
+    F: FnOnce() -> Fut,
+    Fut: std::future::Future<Output = T>,
+{
+    let start = Instant::now();
+    let result = f().await;
+    let duration = start.elapsed();
+
+    trace!(
+        operation = operation_name,
+        duration_ms = duration.as_secs_f64() * 1000.0,
+        "Operation completed"
+    );
+
+    result
+}
+
+// Measures the duration of a database query and logs it using tracing.
+pub async fn measure_db_query<T, F, Fut>(query_name: &str, f: F) -> T
+where
+    F: FnOnce() -> Fut,
+    Fut: std::future::Future<Output = T>,
+{
+    measure_operation(&format!("db_query:{}", query_name), f).await
+}
+
+// Measures the duration of a network request and logs it using tracing.
+pub async fn measure_network_request<T, F, Fut>(request_name: &str, f: F) -> T
+where
+    F: FnOnce() -> Fut,
+    Fut: std::future::Future<Output = T>,
+{
+    measure_operation(&format!("network_request:{}", request_name), f).await
+}
+
+#[cfg(test)]
```

Comment:
I like to test many things in bors, but I think that this is overkill :) Let's not add unnecessary code, this isn't worth it, IMO.